src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -327,6 +329,17 @@ class DetailsCard extends React.Component {
           const operatingSystems = this.props.operatingSystems
           const os = operatingSystems.find(os => os.get('id') === value)
           updates = this.updateOs(updates, os)
+
+          const timeZoneName = updates.getIn(['timeZone', 'name'])
+          const isWindowsTimeZone = timezones.find(timezone => timezone.id === timeZoneName)
+          const isWindowsVm = isWindows(os.get('name'))
+
+          if (isWindowsVm && !isWindowsTimeZone) {
+            changeQueue.push({ fieldName: 'timeZone', value: 'GMT Standard Time' })
+          }
+          if (!isWindowsVm && isWindowsTimeZone) {
+            changeQueue.push({ fieldName: 'timeZone', value: 'Etc/GMT' })
```

Comment:
We should not hard-coded the timezone defaults since they are declared as config values "vdcDefaultGeneralTimeZone", "DefaultWindowsTimeZone" on vdc_options table.
I suggest that in case we want to set the default then set the timeZone to null and let the engine backend choose the value for you as set in https://github.com/oVirt/ovirt-engine/blob/2cc0ab0ff6b00b5b51975497eaf1747b29d2a69c/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/VmHandler.java#L800.

Reply:
absolutely. 
And only send the new default one(from vdc_options indeed) when you switch OS Type to a different family than what's in the template and the template has non-null TZ.